web_poet/requests.py

```diff
@@ -82,18 +84,56 @@ class HttpClient:
     def __init__(self, request_downloader: Callable = None):
         self._request_downloader = request_downloader or _perform_request
 
+    @staticmethod
+    def _handle_status(
+        response: HttpResponse,
+        request: HttpRequest,
+        *,
+        allow_status: List[_Status] = None,
+    ) -> None:
+        allow_status_normalized = list(map(str, as_list(allow_status)))
+        allow_all_status = any(
+            [True for s in allow_status_normalized if "*" == s.strip()]
+        )
+
+        if (
+            allow_all_status
+            or response.status is None  # allows serialized responses from tests
+            or response.status < 400
+            or str(response.status) in allow_status_normalized
+        ):
+            return
+
+        if 400 <= response.status < 500:
+            kind = "Client"
+        elif 500 <= response.status < 600:
+            kind = "Server"
+
+        msg = f"{response.status} {kind} Error for {response.url}"
+        raise HttpResponseError(msg, request=request, response=response)
+
     async def request(
         self,
         url: str,
         *,
         method: str = "GET",
         headers: Optional[_Headers] = None,
         body: Optional[_Body] = None,
+        allow_status: List[_Status] = None,
     ) -> HttpResponse:
-        """This is a shortcut for creating a :class:`~.HttpRequest` instance and executing
-        that request.
+        """This is a shortcut for creating a :class:`~.HttpRequest` instance and
+        executing that request.
 
-        A :class:`~.HttpResponse` instance should then be returned.
+        A :class:`~.HttpResponse` instance should then be returned for successful
+        responses in the 100-3xx status code range. Otherwise, an exception of
+        type :class:`web_poet.exceptions.http.HttpResponseError` will be raised.
```

Comment:
I think it reads like this: successful response with 100-3xx status code? HttpResponse is returned. Otherwise? HttpResponseError. 

It's not entirely correct, because HttpRequestError can also be raised (connection error is not a "successful response with 100-3xx status code").

Reply:
A good catch! Updated the language in https://github.com/scrapinghub/web-poet/pull/38/commits/0eefdf872f4aa9cbf5ad9c52a29626dbbe49e666 to explicitly mention `HttpRequestError` as well.